pkgs/development/compilers/miranda/default.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv, lib, fetchzip, fetchpatch, gcc }:
+
+stdenv.mkDerivation rec {
+  pname = "miranda";
+  version = "2.066";
+
+  # The build generates object files (`.x`) from module files (`.m`).
+  # To be able to invalidate object file, it contains the `mtime`
+  # of the corresponding module file at the time of the build.
+  # When a file is installed to Nix store its `mtime` is set to `1`,
+  # so the `mtime` field in the object file would no longer match
+  # and Miranda would try to regenerate it at the runtime,
+  # even though it is up to date.
+  # Using `fetchzip` will make all the source files have `mtime=1`
+  # from the start so this mismatch cannot occur.
+  src = fetchzip {
+    url = "https://www.cs.kent.ac.uk/people/staff/dat/miranda/src/mira-${builtins.replaceStrings [ "." ] [ "" ] version}-src.tgz";
+    sha256 = "KE/FTL9YW9l7VBAgkFZlqgSM1Bt/BXT6GkkONtyKJjQ=";
+  };
+
+  patches = [
+    # Allow passing `PREFIX` to makeFlags.
+    # Sent upstream on 2020-10-10.
+    (fetchpatch {
+      name = "fix-makefile-variables.patch";
+      url = "https://github.com/jtojnar/miranda/commit/be62d2150725a4c314aa7e3e1e75a165c90be65d.patch";
+      sha256 = "0r8nnr7iyzp1a3w3n6y1xi0ralqhm1ifp75yhyj3h1g229vk51a6";
+    })
+
+    # Create the installation directories.
+    # Sent upstream on 2020-10-10.
+    (fetchpatch {
+      name = "add-mkdirs-makefile.patch";
+      url = "https://github.com/jtojnar/miranda/commit/048754606625975d5358e946549c41ae7b5d3428.patch";
+      sha256 = "1n8xv679i7s789km2dxxrs2pphyyi7vr7rhafqvmkcdmhmxk9h2a";
+    })
+
+    # Use correct installation path for finding the library.
+    # Sent upstream on 2020-10-10.
+    (fetchpatch {
+      name = "c-path-fixes.patch";
+      url = "https://github.com/jtojnar/miranda/commit/aea0a118a802a0da6029b781f7cfd388224263cf.patch";
+      sha256 = "1z3giv8fzc35a23ga9ahz9d1fbvya67kavnb8h4rv2icbzr5j5gd";
+    })
+
+    # Make build reproducible.
+    # Sent upstream on 2020-10-10.
+    (fetchpatch {
+      name = "deterministic-build.patch";
+      url = "https://github.com/jtojnar/miranda/commit/daf8abb8f30ec1cca21698e3fc355578b9f7c571.patch";
+      sha256 = "TC/YrHrMzdlwicJ3oJ/TjwhkufmV3ypemgyqhMmVut4=";
+    })
+  ];
+
+  nativeBuildInputs = lib.optionals stdenv.isDarwin [ gcc ];
+
+  makeFlags = [
+    "CFLAGS=-O2"
```

Comment:
I think this is done in nixpkgs by default by hardening but cannot confirm that at the moment.

Reply:
Without this, building on darwin we have
```
no configure script, doing nothing
building
build flags: -j4 -l4 SHELL=/nix/store/smimdn17mp2ih6nfxwl78mswwdkmcm56-bash-4.4-p23/bin/bash PREFIX=/nix/store/wsf474xndkig4zk8q5ii5npwjd13xl28-miranda-2.066
gcc -w    -c -o big.o big.c
/nix/store/smimdn17mp2ih6nfxwl78mswwdkmcm56-bash-4.4-p23/bin/bash: gcc: command not found
make: *** [<builtin>: big.o] Error 127
builder for '/nix/store/mwpv3hks09q1f4aw406d4jn1acjbymgr-miranda-2.066.drv' failed with exit code 2
error: build of '/nix/store/mwpv3hks09q1f4aw406d4jn1acjbymgr-miranda-2.066.drv' failed
```